src/lexer.c

```diff
@@ -403,26 +475,61 @@ void tryParseExportStatement () {
 
 char16_t readExportAs (char16_t* startPos, char16_t* endPos) {
   char16_t ch = *pos;
+
   if (ch == 'a') {
     pos += 2;
     ch = commentWhitespace(true);
     startPos = pos;
-    readToWsOrPunctuator(ch);
-    endPos = pos;
+
+    if (!isQuote(ch)) {
+      ch = readToWsOrPunctuator(ch);
+      endPos = pos;
+    }
+    // export { mod as "identifer" } from
+    // export { mod as "@notid" } from
+    // export { mod as "spa ce" } from
+    // export { mod as " space" } from
+    // export { mod as "space " } from
+    // export { mod as "not~id" } from
+    // export { mod as "%notid" } from
+    else {
+      startPos = pos + 1;
+
+      if (!scanExportAsQuotes(ch)) {
+        syntaxError();
+        return ch;
+      }
+      readToWsOrPunctuator(ch);
+      endPos = pos - 1;
+    }
```

Comment:
Start pos and end pos here are only being used to check if there was an import string - and not for their actual positions, so there is no need to offset the quotes.

Suggestion:

```suggestion
    // export { mod as "..." } from
    if (isQuote(ch)) {
      stringLiteral(ch);
      ch = readToWsOrPunctuator(ch);
    }
    else {
      ch = readToWsOrPunctuator(ch);
    }
    endPos = pos;
```

Reply:
So that's it (⊙o⊙)… On coding this part, I always want locate the "REAL" position, and I found it works as I **didn't** locate the real position in some cases 😂